src/test/java/com/google/api/codegen/config/ResourceNameMessageConfigsTest.java

```diff
@@ -221,4 +240,130 @@ public void testCreateResourceNameConfigs() {
 
     assertThat(diagCollector.getErrorCount()).isEqualTo(0);
   }
+
+  @Test
+  public void testCreateFlattenings() {
```

Comment:
This method appears to be testing multiple things from multiple classes. Please split into multiple test case classes.

Reply:
The scope of the test is large and covers multiple classes. 

The main reason this test is in this test class is because it uses the the mocks of this class, and I'd rather keep this unit test method here than to copy over the mocks from this class into another test class.

Reply:
Ok that's reasonable. I think it might be nice to change the name of the test class, but probably not important in this PR.